holoviews/element/path.py

```diff
@@ -191,24 +203,52 @@ class Ellipse(BaseShape):
 
     y = param.Number(default=0, doc="The y-position of the ellipse center.")
 
-    height = param.Number(default=1, doc="The height of the ellipse.")
+    height = param.Number(default=1, doc="""
+       The height of the ellipse. Will be deprecated in HoloViews 2.0:
+       use the major_axis_length parameter instead.""")
+
+    major_axis_length = param.Number(default=1, doc="The length of the ellipse major axis")
+
+    minor_axis_length = param.Number(default=1, doc="The length of the ellipse minor axis")
 
-    aspect= param.Number(default=1.0, doc="The aspect ratio of the ellipse.")
+    aspect= param.Number(default=1.0, doc="""
+       Final multipler only applied to the minor axis length. Allows
+       ellipses to be specified without the (major axis length, minor
+       axis length) tuple format.""")
 
-    orientation = param.Number(default=0, doc="Orientation in the Cartesian coordinate system, the counterclockwise angle in radian between the first axis and the horizontal.")
+    orientation = param.Number(default=0, doc="""
+       Orientation in the Cartesian coordinate system, the
+       counterclockwise angle in radians between the first axis and the
+       horizontal.""")
 
     samples = param.Number(default=100, doc="The sample count used to draw the ellipse.")
 
     group = param.String(default='Ellipse', constant=True, doc="The assigned group name.")
 
-    def __init__(self, x, y, height, **params):
-        super(Ellipse, self).__init__([], x=x, y=y, height=height, **params)
+    def __init__(self, x, y, spec, **params):
+
+        if 'height' in params:
+            spec = params['height']
+
+        if isinstance(spec, tuple):
+            (major, minor) = spec
+            if major < minor:
+                raise ValueError('Minor axis length must be less than major axis length')
+            params['height'] = major
+        else:
```

Comment:
Naming parameters for geometrical objects is always difficult because the word that people use depends on things that aren't up to that parameter to decide.  E.g. "height" and "width" are tricky for *anything* with an orientation, since rotating by 90 degrees will make the words seem backwards.  But there's really no way around that, and the alternative of using names with no meaning at all (axis1) isn't any better, because then there's no way for people to guess which dimension will be controlled by which parameter. Given that there is already `height`, I'd suggest just keeping it and adding `width`; why not do the simplest thing, in a case where there is no truly great answer?